SingularityUI/app/controllers/Tail.coffee

```diff
@@ -16,11 +16,14 @@ class TailController extends Controller
             model: @models.taskHistory
 
         app.showView @view
+
         if @offset?
-            @collections.logLines.fetchOffset @offset
+            $.when( @models.taskHistory.fetch(), @collections.logLines.fetchOffset(@offset) ).then =>
+                @view.afterInitialData()
         else
-            @collections.logLines.fetchInitialData()
-
+            $.when( @models.taskHistory.fetch(), @collections.logLines.fetchInitialData() ).then =>
+                @view.afterInitialData()
+        
         @refresh()
```

Comment:
nitpick -- consider storing the fetchOffset / fetchInitialData result as a variable, and then call `$.when(fetchPromise, @refresh())`, so that we're not calling `taskHistory.fetch()` twice